rero_ils/modules/documents/listener.py

```diff
@@ -39,158 +39,178 @@ def enrich_document_data(sender, json=None, record=None, index=None,
     :param index: The index in which the record will be indexed.
     :param doc_type: The doc_type for the record.
     """
-    if index.split('-')[0] == DocumentsSearch.Meta.index:
-        # HOLDINGS
-        holdings = []
-        document_pid = record['pid']
-        es_holdings = HoldingsSearch()\
-            .filter('term', document__pid=document_pid)\
+    if index.split('-')[0] != DocumentsSearch.Meta.index:
+        return
+    holdings = []
+    document_pid = record['pid']
+    es_holdings = HoldingsSearch()\
+        .filter('term', document__pid=document_pid)\
+        .scan()
+    for holding in es_holdings:
+        holding = holding.to_dict()
+        hold_data = {
+            'pid': holding['pid'],
+            'location': {
+                'pid': holding['location']['pid'],
+            },
+            'circulation_category': [{
+                'pid': holding['circulation_category']['pid'],
+            }],
+            'organisation': {
+                'organisation_pid': holding['organisation']['pid'],
+                'library_pid': holding['library']['pid']
+            }
+        }
+        # Index additional holdings fields into the document record
+        holdings_fields = [
+            'call_number', 'second_call_number', 'index',
+            'enumerationAndChronology', 'supplementaryContent',
+            'local_fields'
+        ]
+        for field in holdings_fields:
+            if field in holding:
+                hold_data[field] = holding.get(field)
+        # Index holdings notes
+        if notes := [n['content'] for n in holding.get('notes', []) if n]:
+            hold_data['notes'] = notes
+
+        # Index items attached to each holdings record
+        es_items = ItemsSearch()\
+            .filter('term', holding__pid=holding['pid'])\
             .scan()
-        for holding in es_holdings:
-            holding = holding.to_dict()
-            hold_data = {
-                'pid': holding['pid'],
-                'location': {
-                    'pid': holding['location']['pid'],
-                },
-                'circulation_category': [{
-                    'pid': holding['circulation_category']['pid']
-                }],
-                'organisation': {
+        for item in es_items:
+            item = item.to_dict()
+            item_data = {
+                'pid': item['pid'],
```

Comment:
It can be a good idea to have a variable with the list of fields to process/copy.